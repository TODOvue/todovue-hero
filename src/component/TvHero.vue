<template>
  <div
    class="tv-hero-body"
    :class="{ 'tv-hero-entry': isEntry }"
    v-if="hero"
    :style="custom.customHero"
  >
    <div class="tv-hero" :class="{ 'tv-hero-full': !hero.image }">
      <div class="tv-hero-image" v-if="hero.image">
        <img :src="hero.image" :alt="hero.alt" />
      </div>
      <div class="tv-hero-content">
        <div class="tv-hero-title">
          {{ hero.titleText }}
          <div class="tv-hero-separator" :style="custom.bgAfter"></div>
        </div>
        <div class="tv-hero-description">
          {{ hero.descriptionText }}
        </div>
        <div class="tv-hero-actions" v-if="!isEntry">
          <tv-button
            isRounded
            :customStyle="custom.customButton"
            @click-button="handleClick"
          >
            {{ hero.buttonText }}
          </tv-button>
          <tv-button
            v-if="hero.buttonSecondaryText"
            isRounded
            isOutlined
            :customStyle="custom.customButtonSecondary"
            @click-button="handleClickSecondary"
          >
            {{ hero.buttonSecondaryText }}
          </tv-button>
        </div>
      </div>
    </div>
  </div>
</template>

<script>
import TvButton from "todovue-button";
import useHero from "@/composable/useHero";
export default {
  name: "TvHero",
  props: {
    configHero: {
      type: Object,
      default: () => {},
    },
    customHero: {
      type: Object,
      default: () => {},
    },
    isEntry: {
      type: Boolean,
      default: false,
    },
  },
  components: {
    TvButton,
  },
  setup(props) {
    const { handleClick, handleClickSecondary, custom, hero } = useHero(props);

    return {
      handleClick,
      handleClickSecondary,
      custom,
      hero,
    };
  },
  emits: ["clickButton", "clickSecondaryButton"],
};
</script>

<style scoped lang="scss" src="../assets/scss/style.scss"></style>
