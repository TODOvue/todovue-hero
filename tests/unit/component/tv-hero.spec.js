import { shallowMount } from "@vue/test-utils";
import TvHero from "@/component/TvHero.vue";

describe("TvButton Component", () => {
  const configHero = {
    titleText: "TODOvue Blog",
    descriptionText:
      "Introducing my Vue.js blog! Get ready to dive into the world of Vue.js.",
    buttonText: "View all blogs",
    image: "https://todovue.com/logo.png",
    alt: "TODOvue Logo",
  };

  it("should matches snapshot", () => {
    const wrapper = shallowMount(TvHero, {
      props: {
        configHero,
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should not render card when configCard is null", () => {
    const wrapper = shallowMount(TvHero, {
      props: {
        configHero: null,
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should emmit event when primary button is clicked", async () => {
    const wrapper = shallowMount(TvHero, {
      props: {
        configHero,
      },
    });
    wrapper.vm.handleClick();
    expect(wrapper.emitted("clickButton")).toBeTruthy();
  });

  it("should emmit event when secondary button is clicked", async () => {
    const wrapper = shallowMount(TvHero, {
      props: {
        configHero: {
          ...configHero,
          buttonSecondaryText: "View more",
        },
      },
    });
    wrapper.vm.handleClickSecondary();
    expect(wrapper.emitted("clickSecondaryButton")).toBeTruthy();
  });

  it("should render custom style", () => {
    const wrapper = shallowMount(TvHero, {
      props: {
        configHero: {
          ...configHero,
          buttonSecondaryText: "View last blog",
        },
        customHero: {
          bgBody: "#1e1d23",
          colorBody: "#e1e2dc",
          bgButton: "#8673a1",
          buttonText: "#e1e2dc",
          bgButtonSecondary: "#79308d",
          buttonSecondaryText: "#e1e2dc",
        },
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should return null when customHero is null", () => {
    const wrapper = shallowMount(TvHero, {
      props: {
        configHero,
        customHero: null,
      },
    });
    expect(wrapper.vm.customHero).toEqual(null);
  });
});
