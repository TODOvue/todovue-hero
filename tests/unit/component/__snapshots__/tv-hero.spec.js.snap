// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TvButton Component should matches snapshot 1`] = `
<div class="tv-hero-body">
  <div class="tv-hero tv-hero-full">
    <!--v-if-->
    <div class="tv-hero-content">
      <div class="tv-hero-title">
        <!--v-if-->TODOvue
      </div>
      <div class="tv-hero-description">
        <!--v-if-->TODOvue Blog
      </div>
      <div class="tv-hero-actions">
        <tv-button-stub isrounded="true" isfull="false" isoutlined="false" issmall="false" islarge="false" issuccess="false" icon="" isinfo="false" iswarning="false" iserror="false" isdisabled="false" iscircle="false" istext="false" buttontext="" iconposition="right"></tv-button-stub>
        <tv-button-stub isrounded="true" isfull="false" isoutlined="true" issmall="false" islarge="false" issuccess="false" icon="" isinfo="false" iswarning="false" iserror="false" isdisabled="false" iscircle="false" istext="false" buttontext="" iconposition="right"></tv-button-stub>
      </div>
    </div>
  </div>
</div>
`;
